### DISCUSSION TITLE
Hooks - Get Payload ID in items.update hook

### DISCUSSION THREAD

**Discussion by AlexVanK** [directus/directus#10144] [Archive]
Hello,

sorry if this is a stupid question.
How do I get the current item ID that fires an `items.update` hook?

On
`filter("application_info.items.update", async (input,`
I tried:
`input["id"]`

but the id is not part of the Payload `input`

**Comment by joselcvarela** [comment]
Hello @AlexVanK 
From here https://github.com/directus/directus/blob/76a306e7f434638d14c5bdf7cc946b4b657a614b/api/src/services/items.ts#L400-L413

It seems you can get the primary key on keys, like:
`filter("application_info.items.update", async (input, { keys }`

Give it a try and let us know if that worked 👍

**Reply by AlexVanK** [reply]
This works great!!! Thank you :)

I see that this was flagged for documentation!
Also a good idea, I endorse that!
